core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java

```diff
@@ -18,42 +19,49 @@
 
 import org.apache.flink.api.java.functions.KeySelector;
 
-public class SumOfDelayAtStationFunction {
+public class SumOfDelayAtStationFunction<T extends Object> {
 
     /* This function expects a Stream of TrainDelayNotification events and a window size
     and sums up all delay based on the location Id's in the given window size.
     The window is a fixed event number window.
     It will return a Stream of Tuple2 with the location Id and the sum of delay. 
     */
-    public DataStream<Tuple2<Long, Double>> SumOfDelayAtStation(DataStream<TrainDelayNotification> inputStream, int windowSize) {
+    public DataStream<Tuple2<Long, Double>> SumOfDelayAtStation(DataStream<T> inputStream, int windowSize, String locationAttributName) {
         DataStream<Tuple2<Long, Double>> resultStream = inputStream
         .keyBy(
-            new KeySelector<TrainDelayNotification, Integer>(){
+            new KeySelector<T, Integer>(){
                 Integer key = Integer.valueOf(0);
-                public Integer getKey(TrainDelayNotification event){
+                public Integer getKey(T event){
                     Integer returnKey = key/windowSize;
                     key++;
                     return returnKey;
-                } 
+                }
             }
         )
         .window(GlobalWindows.create()).trigger(
             CustomCountTrigger.of(windowSize)
         ).process(
-            sumOfDelayAtStationWindowProcessFunction()
+            sumOfDelayAtStationWindowProcessFunction(locationAttributName)
         );
         return resultStream;
-    }; 
+    };
 
-    public static ProcessWindowFunction<TrainDelayNotification, Tuple2<Long, Double>, Integer, GlobalWindow> sumOfDelayAtStationWindowProcessFunction() {
-        return new ProcessWindowFunction<TrainDelayNotification, Tuple2<Long, Double>, Integer, GlobalWindow>() {
+    public ProcessWindowFunction<T, Tuple2<Long, Double>, Integer, GlobalWindow> sumOfDelayAtStationWindowProcessFunction(String locationAttributName) {
```

Comment:
As I understand it, the Tuple<Long, Double> is an "event" that returns <stationId, sumOfDelays>. Why is the stationId as a Long? They are given as integers in our Protobufs

Reply:
Somewhere it occured that the stationId must be a Long so I went with it.